### PR TITLE
ColorPicker: Adding Preview box to the ColorPicker component

### PR DESCRIPTION
**Pull request by navkuma**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

1. Adding the  preview pane to the color picker component
2. Increase  the width and height of the color rectangle
3. CSS changes 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10264)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx [hunk 1]
```diff
@@ -99,22 +100,55 @@ export class ColorPickerBase extends React.Component<IColorPickerProps, IColorPi
...
-              className="is-alpha"
-              isAlpha
-              overlayStyle={{ background: `linear-gradient(to right, transparent 0, #${color.hex} 100%)` }}
-              minValue={0}
-              maxValue={MAX_COLOR_ALPHA}
-              value={color.a}
-              onChange={this._onAChanged}
-            />
-          )}
+          <ColorRectangle color={color} onChange={this._onSVChanged} styles={{ root: { height: 268 } }} />
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx]: Valid point, I am now moving this up and passing it as pros from Color Picker

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.styles.ts [hunk 2]
```diff
@@ -9,14 +9,15 @@ export const getStyles = (props: IColorPickerStyleProps): IColorPickerStyles =>
       theme.fonts.medium,
       {
         position: 'relative',
-        maxWidth: 300
+        maxWidth: 352,
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.styles.ts]: Done

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts [hunk 3]
```diff
@@ -80,6 +80,12 @@ export interface IColorPickerProps {
    * Call to provide customized styling that will layer on top of the variant rules.
    */
   styles?: IStyleFunctionOrObject<IColorPickerStyleProps, IColorPickerStyles>;
+
+  /**
+   * whether color preview box is visible.
+   * @defaultvalue false
+   */
+  previewBoxVisible?: boolean;
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: Done

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts [hunk 4]
```diff
@@ -80,6 +80,12 @@ export interface IColorPickerProps {
    * Call to provide customized styling that will layer on top of the variant rules.
    */
   styles?: IStyleFunctionOrObject<IColorPickerStyleProps, IColorPickerStyles>;
+
+  /**
+   * whether color preview box is visible.
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: changed

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx [hunk 5]
```diff
@@ -99,22 +100,55 @@ export class ColorPickerBase extends React.Component<IColorPickerProps, IColorPi
...
+          <table className={classNames.table} cellPadding="0" cellSpacing="0">
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx]: Done, I have used Flex.

**Comment by navkuma** [comment]
> Thanks for the contrib! Please do 2 things:
> 
> 1. you should link your github account with your MSFT here: https://opensource.microsoft.com/
> 2. it would be helpful (since you didn't have a issue backing this PR) to have an added example in the colorpicker component

Done, Linked my github account to MSFT and added the preview box to example as well !!

**Comment by navkuma** [comment]
> Needs more details to proceed. @Jahnp - just a ping of a desire to add a preview for colorpicker.
> 
> @navkuma - There's actually a Sketch file that has the design of these pickers. Please work with Peter & design team on new component features like these.
> 
> https://aka.ms/FluentToolkits/Web/Sketch
> 
> image

Color Picker Updated Design Spec Link: https://microsoft.sharepoint.com/:p:/t/powerpoint/EWzqUHiPa_1Cqb7x5EEvFbYBDenpYaopyWGFnTbBGMat_g?rtime=qoN5dcwb10g

**Comment by vaverma** [comment]
Hi @ecraig12345 , I have addressed all the comments. Kindly review them. Also, Can you tell me why Size Auditor checks are failing? Seems like the failures are not related to our changes.

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx [hunk 6]
```diff
@@ -99,22 +100,63 @@ export class ColorPickerBase extends React.Component<IColorPickerProps, IColorPi
       className
     });
 
+    const classNamesPreviewBox = mergeStyleSets({
```
[the remark this answers is not in the document]
Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx]: Done, Now these styles are generated as part of ColorPicker.styles.ts

**Comment by vaverma** [comment]
Can some one please tell me why these Tests are failing ?

**Comment by vaverma** [comment]
Can someone please help me figure out what needs to be done to merge this PR?

**Comment by kenotron** [comment]
i see you're encountering issues with the snapshots. Please run:

```
yarn build --min --no-lint
lerna run update-snapshots
```

this should build all the packages (takes quite a few minutes), then will run the update-snapshots script on all the packages

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts [hunk 7]
```diff
@@ -80,6 +80,17 @@ export interface IColorPickerProps {
    * Call to provide customized styling that will layer on top of the variant rules.
    */
   styles?: IStyleFunctionOrObject<IColorPickerStyleProps, IColorPickerStyles>;
+
+  /**
+   * Call to provide customized styling for Preview Box.
+   */
+  rectangleStyle?: IStyleFunctionOrObject<IColorPickerStyleProps, IColorPickerStyles>;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: Can you remove this one? It does not follow any other convention and doesn't seem to be needed.

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: Just catching up on the full conversation here. It seems like the intent is to pass in style overrides to the rectangle. Looking at the code to understand if there's a better way

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: So it seems like the props are wrong anyway here. It probably should be:

```tsx
rectangleStyles: IStyleFunctionOrObject<IColorRectangleStyleProps, IColorRectangleStyles>
```
But, I'd prefer to remove them. Is there a real need here? Can you clarify why?

Comment (vaverma) [reply packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts]: Ok I have removed it now, and I am using colorRectangle Style to pass the height of the rectangle.  Can you now help me merging the PR

File: packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.md [hunk 8]
```diff
@@ -2527,7 +2527,9 @@ export interface IColorPickerProps {
     greenLabel?: string;
     hexLabel?: string;
     onChange?: (ev: React.SyntheticEvent<HTMLElement>, color: IColor) => void;
+    rectangleStyle?: IStyleFunctionOrObject<IColorPickerStyleProps, IColorPickerStyles>;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.md]: should get removed if you run `update-api`

Comment (vaverma) [reply packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.md]: done

**Pull request merged**